src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -315,14 +315,23 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
         configmap_rg_name = configmap.data["AZURE_RESOURCE_GROUP"]
         configmap_cluster_name = configmap.data["AZURE_RESOURCE_NAME"]
         if connected_cluster_exists(client, configmap_rg_name, configmap_cluster_name):
+            preview_cluster_resource = None
+            public_key = None
+
+            try:
+                preview_cluster_resource = get_connectedk8s_2023_11_01(cmd, configmap_rg_name,
```

Comment:
>get_connectedk8s_2023_11_01 [](http://example.com/codeflow?start=43&length=27)

If the cluster doesn't exist, this will return null, right?  So, check the value of preview_cluster_resource, before using its properties.  Do the same in all the places where you are calling get_connectedk8s_2023_11_01.  There are 4 or 5 more instances

Reply:
If the cluster doesn't exist, this will raise an error with the failing HTTTP status, not just return null.
It looks like this:
<img width="1118" alt="image" src="https://github.com/Azure/azure-cli-extensions/assets/26911860/722506fe-5e43-46f7-b444-99b4b9345a7d">

Reply:
Have you confirmed that it errors out when invoked from within the CLI too?  If so, yes, you don't need to check for null.  In that case, is that error message thrown sufficient for the user?

Reply:
Yes, the screenshot I shared above was me invoking that code path for the connectedk8s show subcommand with a private build of the connectedk8s extension containing my changes.
I think the error message is sufficient enough to convey to the user that the referenced cluster resource doesn't exist.

Reply:
Yes, that screenshot I shared above was from when I invoked that code path via CLI locally - this was done with a private build of the connectedk8s extension containing my changes.
I think the message thrown is sufficient, it is similar behavior as what it was previously.